config/config.go

```diff
@@ -40,29 +35,19 @@ type UpstreamConf struct {
 	PullingError error
 }
 
-func (uc *UpstreamConf) Hash() string {
-	var kv [][2]string
+func (uc *UpstreamConf) Equal(that *UpstreamConf) bool {
```

Comment:
Uh... I guess you don't have to write your own DeepEqual.
There is a fancy impl from Go standard library `reflect.DeepEqual`

Reply:
... though I didn't test the performance ....

Reply:
I think there maybe is no need because the struct of map is not complex and DeepEqual is slower than the func I write here obviously.

Reply:
alright...